src/browser/chrome.ts

```diff
@@ -157,8 +191,40 @@ export const openUrl = async ({
         msg.type() === "error" &&
         log(`JS error |`, msg.text().replace(/\n[\w\W]*/, ""), `@ [${url}]`)
     );
+
+    let counterRequestsUnfinished = 0;
+    const updateCounter = async (counter: number) => { 
+      counterRequestsUnfinished = counter;
+      try {
+        await page.evaluate(({ counterRequestsUnfinished, COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME }) => {
+          const item = counterRequestsUnfinished.toString();
+          localStorage.setItem(COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME, item);
+        }, { counterRequestsUnfinished, COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME });
+      } catch (e) {}
+    };
+    const onRequestProcessed = async () => {
+      try {
+        await page.evaluate(({ LAST_RESPONSE_AT_VARIABLE_NAME }) => {
+          const item = Date.now().toString();
+          localStorage.setItem(LAST_RESPONSE_AT_VARIABLE_NAME, item);
+        }, {LAST_RESPONSE_AT_VARIABLE_NAME});
+      } catch (e) {}
+     };
+
+    page.on('request', async () => {
+      await updateCounter(counterRequestsUnfinished + 1);
+    });
+    page.on('requestfailed', async () => { 
+      await updateCounter(counterRequestsUnfinished - 1);
+      await onRequestProcessed();
```

Comment:
Instead of 2 functions and evaluates, you could have just one function with a flag, for instance, 

```ts
await updateCounter(counterRequestsUnfinished - 1, {
  requestEnd: true
});
```

or even more smarter,

```ts
await updateCounter(-1);
```

That's all you need. And the function would abstract the rest (inc/dec + setting vars). 

Please simplify